smartsim/_core/utils/telemetry/telemetry.py

```diff
@@ -0,0 +1,573 @@
+        and local launching
+
+        :param launcher: the name of the workload manager used by the experiment
+        :type launcher: str
+        :raises SSUnsupportedError: if a string is passed that is not
+        a supported launcher
+        :raises TypeError: if no launcher argument is provided."""
+        if not launcher:
+            raise TypeError("Must provide a 'launcher' argument")
+
+        if launcher_type := self._launcher_map.get(launcher.lower(), None):
+            return launcher_type()
+
+        raise ValueError("Launcher type not supported: " + launcher)
+
+    def set_launcher(self, launcher: str) -> None:
+        """Set the launcher for the experiment
+
+        :param launcher: the name of the workload manager used by the experiment
+        :type launcher: str"""
+        self._launcher = self.init_launcher(launcher)
+        self.job_manager.set_launcher(self._launcher)
+        self.job_manager.start()
+
+    def process_manifest(self, manifest_path: str) -> None:
+        """Read the manifest for the experiment. Process the
+        `RuntimeManifest` by updating the set of tracked jobs
+        and registered collectors
+
+        :param manifest_path: full path to the manifest file
+        :type manifest_path: str"""
+        try:
+            # it is possible to read the manifest prior to a completed
+            # write due to no access locking mechanism. log the issue
+            # and continue. it will retry on the next event loop iteration
+            manifest = RuntimeManifest.load_manifest(manifest_path)
+            if not manifest:
+                return
+        except json.JSONDecodeError:
+            logger.error(f"Malformed manifest encountered: {manifest_path}")
+            return
+        except ValueError:
+            logger.error("Manifest content error", exc_info=True)
+            return
+
+        if self._launcher is None:
+            self.set_launcher(manifest.launcher)
+
+        if not self._launcher:
+            raise SmartSimError(f"Unable to set launcher from {manifest_path}")
+
+        # filter out previously added items
+        runs = [run for run in manifest.runs if run.timestamp not in self._tracked_runs]
+
+        # manifest is stored at <exp_dir>/.smartsim/telemetry/manifest.json
+        exp_dir = pathlib.Path(manifest_path).parent.parent.parent
+
+        for run in runs:
+            for entity in run.flatten(
+                filter_fn=lambda e: e.key not in self._tracked_jobs
+            ):
+                entity.path = str(exp_dir)
+
+                # track everything coming in (managed and unmanaged)
+                self._tracked_jobs[entity.key] = entity
+
+                # register collectors for new entities as needed
+                if entity.telemetry_on:
+                    self._collector_mgr.register_collectors(entity)
+
+                # persist a `start` event for each new entity in the manifest
+                write_event(
```

Comment:
Small question: For unmanaged tasks would this write_event be overwriting what indirect.py has written?  I'm wondering if this should just be for unmanaged tasks.

Reply:
It is the same, though `write_event` will purposefully not overwrite an existing record. Because it doesn't change the old file, I just left this call conditional-free. I can add it if you want it.

Reply:
I added a test that should demonstrate the non-destructive call. See:

https://github.com/CrayLabs/SmartSim/pull/460/commits/6276e08ec966a6ede1bc193f83d7bbc1a0f9effc